cv_next/components/layout/navbar/popup.tsx

```diff
@@ -5,38 +5,89 @@ import closeIcon from "@/public/images/closeIcon.png";
 import { useRef, useEffect } from "react";
 import Link from "next/link";
 import { useSupabase } from "@/hooks/supabase";
+import profileIcon from "@/public/images/profile.png";
+import { useTheme } from "next-themes";
 
 const navLinks = [
   {
     route: "Login",
     path: "/login",
+    req_login: false,
+  },
+  {
+    route: "Feed",
+    path: "/feed",
+    req_login: true,
   },
   {
     route: "Signout",
     path: "/",
+    req_login: true,
   },
 ];
 
 interface PopupProps {
   closeCb: () => void;
+  userData: UserModel | null;
+  updateSignOut: () => void;
 }
 
-export default function Popup({ closeCb }: PopupProps) {
+export default function Popup({
+  closeCb,
+  userData,
+  updateSignOut,
+}: PopupProps) {
   const dialogRef = useRef<HTMLDialogElement>(null);
+  const supabase = useSupabase();
+  const { theme } = useTheme();
+
   useEffect(() => {
     if (dialogRef.current) {
       dialogRef.current.show();
     }
   }, []);
-  const supabase = useSupabase();
 
   const handleSelection = (route: string) => {
     if (route === "Signout") {
       supabase.auth.signOut();
+      updateSignOut();
     }
     closeCb();
   };
 
+  const matchThemePlaceholderImage =
+    theme == "dark" ? { filter: "invert(0)" } : { filter: "invert(1)" };
+
+  const userDataComponent = userData ? (
```

Comment:
The `Popup` component is starting to get a bit too messy.
Quick suggestion to help maintain it: separate this `UserDataComponent` to a different component (can still be here on the same page).